### PR TITLE
Added support for File System root directory

### LINKED ISSUE #9319: Cannot get or set ACL of Data Lake Gen2 filesystem root.
I tried to get & set ACL of ADLS Gen2 filesystem root, but it doesn't work.

I tried the code below.
```code.cs
var sharedKeyCredential = new StorageSharedKeyCredential(storageAccountName, storageAccountKey);
var serviceClient = new DataLakeServiceClient(new Uri($"https://{storageAccountName}.dfs.core.windows.net"), sharedKeyCredential);
var fileSystemClient = serviceClient.GetFileSystemClient(containerName);
var directoryClient = fileSystemClient.GetDirectoryClient("");
var res = await directoryClient.GetAccessControlAsync();
```

This code throws the exception below.

Azure.RequestFailedException
ErroCode: 400
Message: `Value for one of the query parameters specified in the request URI is invalid.`
Stack trace:
```
   at Azure.Storage.Files.DataLake.DataLakeRestClient.Path.GetPropertiesAsync_CreateResponse(Response response)
   at Azure.Storage.Files.DataLake.DataLakeRestClient.Path.GetPropertiesAsync(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, Uri resourceUri, String requestId, Nullable`1 timeout, Nullable`1 action, Nullable`1 upn, String leaseId, Nullable`1 ifMatch, Nullable`1 ifNoneMatch, Nullable`1 ifModifiedSince, Nullable`1 ifUnmodifiedSince, Boolean async, String operationName, CancellationToken cancellationToken)
   at Azure.Storage.Files.DataLake.DataLakePathClient.GetAccessControlInternal(Nullable`1 userPrincipalName, DataLakeRequestConditions conditions, Boolean async, CancellationToken cancellationToken)
   at Azure.Storage.Files.DataLake.DataLakePathClient.GetAccessControlAsync(Nullable`1 userPrincipalName, DataLakeRequestConditions conditions, CancellationToken cancellationToken)
   at b_ridgeAnalyticDbFunctions.Function.AddAclTest.Run(HttpRequest req, ILogger log) in C:\Users\masukawa\source\repos\b-ridgeAnalyticDbFunctions\b-ridgeAnalyticDbFunctions\Function\AddAclTest.cs:line 32
```

It seems send the request below.
```HEAD https://{storageAccountName}.dfs.core.windows.net/{containerName}?action=getAccessControl```

Via azure rest API, I can get ACL of filesystem root by the request below.
```HEAD https://{storageAccountName}.dfs.core.windows.net/{containerName}/?action=getAccessControl```

The difference is a slash after the container name.
How can I do this by the SDK?

### PR DESCRIPTION
**Pull request by seanmcc-msft**
Resolves https://github.com/Azure/azure-sdk-for-net/issues/9319

### CODE REVIEW

**Review by amnguye — APPROVED** [review]

File: sdk/storage/Azure.Storage.Common/src/Shared/UriExtensions.cs [hunk 1]
```diff
@@ -23,6 +23,12 @@ internal static class UriExtensions
         public static Uri AppendToPath(this Uri uri, string segment)
         {
             var builder = new UriBuilder(uri);
+
+            if (segment == "/")
```
[the remark this answers is not in the document]
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Common/src/Shared/UriExtensions.cs]: Nope, we don't need it.  Removed.

File: sdk/storage/Azure.Storage.Files.DataLake/tests/DirectoryClientTests.cs [hunk 2]
```diff
@@ -489,6 +489,22 @@ public async Task GetAccessControlAsync()
...
+
+            // Act
+            PathAccessControl accessControl = await directory.GetAccessControlAsync();
+
+            // Assert
+            Assert.IsNotNull(accessControl.Owner);
+            Assert.IsNotNull(accessControl.Group);
+            Assert.IsNotNull(accessControl.Permissions);
+            Assert.IsNotNull(accessControl.AccessControlList);
+        }
```
[the remark this answers is not in the document]
Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.DataLake/tests/DirectoryClientTests.cs]: Added.

**Review by JoshLove-msft — APPROVED** [review]

**Comment by GBBBAS** [comment]
Very much need this feature - do we know when the next nuget release is expected with this fix?

**Comment by tg-msft** [comment]
https://www.nuget.org/packages/Azure.Storage.Files.DataLake/12.0.0-preview.9

**Pull request merged**